File: src/api/Coursework.Domain/Entities/Vacancy.cs
namespace Coursework.Domain.Entities
{
    public class Vacancy
    {
        public Guid GID { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public int Salary { get; set; }

        public string Location { get; set; }

        public List<string> Skills { get; set; }

        public Guid VacancyStatusGID { get; set; }

        public Guid RecruiterGID { get; set; }

        public VacancyStatus VacancyStatus { get; set; }

        public Recruiter Recruiter { get; set; }
    }
}

File: src/frontend/src/app/vacancy/models/vacancy.ts
import { Skill } from "../../candidates/models/skill";

export class Vacancy {
  public name: string | null = null;
  public description: string | null = null;
  public salary: number | null = null;
  public location: string | null = null;
  public skills: Array<Skill> | null = null;
  vacancyStatusGID: string | null = null;
  recruiterGID: string | null = null;

  constructor(name: string,
              description: string,
              salary: number,
              location: string,
              skills: Array<Skill>,
              vacancyStatusGID: string,
              recruiterGID: string) {
    this.name = name;
    this.description = description;
    this.salary = salary;
    this.location = location;
    this.skills = skills;
    this.vacancyStatusGID = vacancyStatusGID;
    this.recruiterGID = recruiterGID;
  }
}
